GitCommands/AsyncLoader.cs

```diff
@@ -126,6 +126,17 @@ private bool OnLoadingError(Exception exception)
             LoadingError(this, args);
             return args.Handled;
         }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_cancelledTokenSource != null)
```

Comment:
:exclamation: You also need to check the previous value of `_cancellationTokenSource` at the point it is assigned in `Load()` above, and call `Dispose()` before discarding the old reference.